include/thread_pool/thread_pool_options.hpp

```diff
@@ -2,21 +2,77 @@
 
 #include <algorithm>
 #include <thread>
+#include <functional>
+#include <chrono>
 
 namespace tp
 {
 
 /**
- * @brief The ThreadPoolOptions class provides creation options for
- * ThreadPool.
+ * @brief The ThreadPoolOptions class provides creation options for ThreadPool.
  */
 class ThreadPoolOptions
 {
 public:
+    /**
+    * @brief The BusyWaitOptions class provides worker busy wait behaviour options.
+    */
+    class BusyWaitOptions
+    {
+    public:
+        using IterationFunction = std::function<std::chrono::microseconds(size_t)>;
+
+        /**
+        * @brief BusyWaitOptions Construct default options for busy wait behaviour.
+        */
+        BusyWaitOptions(size_t num_iterations = defaultNumIterations()
+            , IterationFunction&& function = defaultIterationFunction());
+
+        /**
+        * @brief setNumIterations Set the number of sleeping iterations that take place during
+        * a worker's busy wait state. The iteration function will be called on every iteration 
+        * with the iteration number.
+        * @param count The number of busy wait iterations.
+        */
+        void setNumIterations(size_t count);
+
+        /**
+        * @brief setIterationFunction Set the function to be called upon each sleep iteration.
+        * @param function The iteration function to be called.
+        */
+        void setIterationFunction(IterationFunction&& function);
```

Comment:
I'm not super convinced about the rvalue references everywhere here, will revisit.